src/alpaka/AlpakaCore/alpakaMemory.h

```diff
@@ -73,21 +75,68 @@ namespace cms::alpakatools {
   template <typename T>
   using host_buffer = typename detail::buffer_type<DevHost, T>::type;
 
+  // non-cached, scalar and 1-dimensional host buffers
+  // the memory is pinned explicitly
+
   template <typename T>
   std::enable_if_t<not std::is_array_v<T>, host_buffer<T>> make_host_buffer() {
-    return alpaka::allocBuf<T, Idx>(host, Scalar{});
+    auto buffer = alpaka::allocBuf<T, Idx>(host, Scalar{});
+    alpaka::prepareForAsyncCopy(buffer);
+    return buffer;
   }
 
   template <typename T>
   std::enable_if_t<cms::is_unbounded_array_v<T> and not std::is_array_v<std::remove_extent_t<T>>, host_buffer<T>>
   make_host_buffer(Extent extent) {
-    return alpaka::allocBuf<std::remove_extent_t<T>, Idx>(host, Vec1D{extent});
+    auto buffer = alpaka::allocBuf<std::remove_extent_t<T>, Idx>(host, Vec1D{extent});
+    alpaka::prepareForAsyncCopy(buffer);
+    return buffer;
   }
 
   template <typename T>
   std::enable_if_t<cms::is_bounded_array_v<T> and not std::is_array_v<std::remove_extent_t<T>>, host_buffer<T>>
   make_host_buffer() {
-    return alpaka::allocBuf<std::remove_extent_t<T>, Idx>(host, Vec1D{std::extent_v<T>});
+    auto buffer = alpaka::allocBuf<std::remove_extent_t<T>, Idx>(host, Vec1D{std::extent_v<T>});
+    alpaka::prepareForAsyncCopy(buffer);
+    return buffer;
+  }
+
+  // potentially cached, scalar and 1-dimensional host buffers, associated to an work queue
+  // the memory is pinned by the caching allocator, or explicitly if it is not used
+
+  template <typename T, typename TQueue>
+  std::enable_if_t<not std::is_array_v<T>, host_buffer<T>> make_host_buffer(TQueue const& queue [[maybe_unused]]) {
+    if constexpr (allocator_policy<alpaka::Dev<TQueue>> == AllocatorPolicy::Caching) {
+      return allocCachedBuf<T, Idx>(host, queue, Scalar{});
+    } else {
+      auto buffer = alpaka::allocBuf<T, Idx>(host, Scalar{});
+      alpaka::prepareForAsyncCopy(buffer);
+      return buffer;
```

Comment:
How about
```suggestion
      return make_host_buffer<T>();
```
to make it explicit that the code is intended to be the same? (same for the other two cases below)

Reply:
mhm... I think I need to wrok a bit more on the combination of all the cases:
  - accelerator GPU memory (sync, async, cached) with a GPU accelerator queue
  - accelerator GPU memory (sync, async ?, cached ?) without a queue
  - accelerator CPU memory¹ (sync, async, cached) with a CPU accelerator queue
  - accelerator CPU memory¹ (sync, async ?, cached ?) without a queue
  - pinned host CPU memory (sync, async? cached) with a GPU accelerator queue
  - pinned host CPU memory (sync, async? cached?) without a queue
  - (pinned) host CPU memory (sync, async? cached?) with a CPU accelerator queue
  - (pinned) host CPU memory (sync, async? cached?) without a queue

¹ is for the case where the backend is a CPU accelerator, like TBB or serial; in these case what can we assume about race conditions between host and "device" memory access ?

Reply:
Splitting `getCachingAllocator` into `getHostCachingAllocator` and `getDeviceCachingAllocator` should help here as well.

Reply:
> How about [...] to make it explicit that the code is intended to be the same? (same for the other two cases below)

OK, after the splitting of the host and device allocators,I think I've followed your suggestion for the fallback host allocations.